addons/project/models/project.py

```diff
@@ -1536,7 +1536,8 @@ def copy(self, default=None):
         if self.recurrence_id:
             default['recurrence_id'] = self.recurrence_id.copy().id
         if self.allow_subtasks:
-            default['child_ids'] = [child.copy({'name': child.name} if has_default_name else None).id for child in self.child_ids]
+            default['child_ids'] = [
+                child.copy({**({'name': child.name} if has_default_name else {}), 'stage_id': child.stage_id.id}).id for child in self.child_ids]
```

Comment:
I finally investigate and indeed we have to copy the stage_id but only when we copy the stage_id for the parent task, that is, in your use case, it is only the case when we duplicate the project and not all the time since `stage_id` field has `copy=False` in its definition.

So, I would suggest doing that:

```diff
diff --git a/addons/project/models/project.py b/addons/project/models/project.py
index 23c6e4a7c980..48716c242538 100644
--- a/addons/project/models/project.py
+++ b/addons/project/models/project.py
@@ -1538,8 +1538,16 @@ class Task(models.Model):
         if self.recurrence_id:
             default['recurrence_id'] = self.recurrence_id.copy().id
         if self.allow_subtasks:
-            default['child_ids'] = [
-                child.copy({**({'name': child.name} if has_default_name else {}), 'stage_id': child.stage_id.id}).id for child in self.child_ids]
+            default_child_ids = []
+            should_copy_stage_id = bool(default.get('stage_id', False))
+            for child in self.child_ids:
+                subtask_default = {}
+                if has_default_name:
+                    subtask_default['name'] = child.name
+                if should_copy_stage_id:
+                    subtask_default['stage_id'] = child.stage_id.id
+                default_child_ids.append(child.copy(subtask_default).id)
+            default['child_ids'] = default_child_ids
         task_copy = super(Task, self).copy(default)
         if self.allow_task_dependencies:
             task_mapping = self.env.context.get('task_mapping')
```